src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -710,24 +702,24 @@ public void GroveI2cStart(GrovePort port, byte addr, byte[] arrayToSend, byte in
             }
             else
             {
-                throw new ArgumentException($"{nameof(GroveI2cStart)} error: Port unsupported. Must be either Grove 1 or Grove 2.");
+                throw new ArgumentException(nameof(port), $"Port unsupported. Must be either {nameof(GrovePort.Grove1)} or {nameof(GrovePort.Grove2)}.");
```

Comment:
Is state the result of a null check?

Reply:
This would only execute if the two if checks above failed I Suspect they would have NPEd already so really this is *NOT* an `ArgumentNullException`, my error. That said, do we want/need to protect against a null `port` argument value up above line 393?

Reply:
`port` is an enum type. I'm not sure what the `null` rules are on that.

Reply:
Good point, since it's an Enum, it's a value-type... at worst it's an invalid value, so this should stay an `ArgumentException`, please disregard this suggestion